src/Adapters/Auth/AuthAdapter.js

```diff
@@ -1,20 +1,61 @@
 /*eslint no-unused-vars: "off"*/
 export class AuthAdapter {
-  /*
+  constructor() {
+    /**
+     * If set to true the signup/login
+     * will fail if user do not provide
+     * authData associated to this Auth Adapter
+     */
+    this.required = false;
+
+    /**
+     * Set to true if you want to force to validateAuthData
+     * even if authData do not change
+     */
+    this.alwaysValidate = false;
```

Comment:
When should the validation occur?

Reply:
Currently validation only occur if authData is detected as changed (a loadash equality check on the DB authData and provided authData)

Reply:
The option allow to force the validation even if authData are detected as same

Reply:
But what always mean? Every time that the user logs in, every time that the session token is used?

Reply:
every time when authData is touched by the client ! (login/save/signup)

Reply:
I believe that the token should be always validated at login regardless of this option. Otherwise we can have a security issue since someone would be able to use the current token, even if expired, to log in. What is the use case that you imagined for always validate false?

Reply:
> Otherwise we can have a security issue since someone would be able to use the current token,

Parse Server has always handled authData this way. i think also like you, it's a security issue ! So i will remove the alwaysValidate options, remove the loadash equality check, and the `hasMutatedAuthData` behavior

Reply:
@davimacedo  @TomWFox may be we should open a security advisory about this